develop/items/custom-data-components.md

```diff
@@ -0,0 +1,267 @@
+---
+title: Custom Data Components
+description: Learn how to add custom data to your items using the new 1.21 component system.
+authors:
+  - Romejanic
+---
+
+# Custom Data Components {#custom-data-components}
+
+As your items grow more complex, you may find yourself needing to store custom data associated with your item. The game allows you to store persistent data within an `ItemStack`, and as of 1.21 the way we do that is by using **Data Components**.
```

Comment:
This was introduced in 1.20.5, not 1.21. Not sure how relevant it is anyway.

Reply:
My bad, I thought they were added in 1.21